src/savegame_json.cpp

```diff
@@ -1675,6 +1675,21 @@ void faction::deserialize(JsonIn &jsin)
     }
     jo.read("size", size);
     jo.read("power", power);
+    if ( !jo.read( "combat_ability", combat_ability )){
+        combat_ability = 100;
+    }else{
+        jo.read("combat_ability", combat_ability);
+    }
+    if ( !jo.read( "food_supply", food_supply )){
+        food_supply = 100;
+    }else{
+        jo.read("food_supply", food_supply);
+    }
+    if ( !jo.read( "wealth", wealth )){
+        wealth = 100;
+    }else{
+        jo.read("wealth", wealth);
```

Comment:
If `JsonObject::read` returns true, the value has been read and put into the variable, there is no point in calling it again.